crates/net/network/src/network.rs

```diff
@@ -460,26 +444,12 @@ pub(crate) enum NetworkHandleMessage<N: NetworkPrimitives = EthNetworkPrimitives
     DisconnectPeer(PeerId, Option<DisconnectReason>),
     /// Broadcasts an event to announce a new block to all nodes.
     AnnounceBlock(NewBlock<N::Block>, B256),
-    /// Sends a list of transactions to the given peer.
-    SendTransaction {
-        /// The ID of the peer to which the transactions are sent.
-        peer_id: PeerId,
-        /// The shared transactions to send.
-        msg: SharedTransactions<N::BroadcastedTransaction>,
-    },
-    /// Sends a list of transaction hashes to the given peer.
-    SendPooledTransactionHashes {
-        /// The ID of the peer to which the transaction hashes are sent.
-        peer_id: PeerId,
-        /// The new pooled transaction hashes to send.
-        msg: NewPooledTransactionHashes,
-    },
     /// Sends an `eth` protocol request to the peer.
-    EthRequest {
-        /// The peer to send the request to.
+    EthMessage {
+        /// The peer to send the message to.
         peer_id: PeerId,
-        /// The request to send to the peer's sessions.
-        request: PeerRequest<N>,
+        /// The message to send to the peer's sessions.
+        message: PeerMessage<N>,
     },
```

Comment:
this should be an additional variant, then this becomes a lot easier, we can still unify those but I'd like to have this as an additional variant